### PR TITLE
feat: add `Parse.Cloud.afterFind(Parse.File)`

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Currently, there are no triggers for file gets.

Related issue: #6572, #7572, #7928

### Approach

**Adds preferred syntax Parse.Cloud.beforeSave(Parse.File)**

The file triggers were made to have their own function names, which adds unnecessary complexity. This allows for `Parse.Cloud.beforeSave(Parse.File)`, `Parse.Cloud.afterSave(Parse.File)`, `Parse.Cloud.beforeDelete(Parse.File)`, `Parse.Cloud.afterDelete(Parse.File)`. In a future version, the current file triggers should be deleted.

**Adds Parse.Cloud.beforeCreate(Parse.File)**

This can only be used on `Parse.File` for the meantime, but in the future could be added to detect class creation. This triggers when a file is expanded, allowing mutation of the file url.

**Adds Parse.Cloud.beforeFind**

This allows a file trigger on the get request of a file. req.download can be set as well to allow file downloads.

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by dblythy** [comment]
This is still a bit of a proof of concept at this stage, more tests will need to be added prior to merge

File: DEPRECATIONS.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ The following is a list of deprecations, according to the [Deprecation Policy](h
 | DEPPS2 | Config option `directAccess` defaults to `true` | [#6636](https://github.com/parse-community/parse-server/pull/6636)   | 5.0.0 (2022)                    | 6.0.0 (2023)                    | deprecated            | -     |
 | DEPPS3 | Config option `enforcePrivateUsers` defaults to `true` | [#7319](https://github.com/parse-community/parse-server/pull/7319)   | 5.0.0 (2022)                    | 6.0.0 (2023)                    | deprecated            | -     |
 | DEPPS4 | Remove convenience method for http request `Parse.Cloud.httpRequest` | [#7589](https://github.com/parse-community/parse-server/pull/7589)   | 5.0.0 (2022)                    | 6.0.0 (2023)                    | deprecated            | -     |
+| DEPPS5 | Remove file triggers in preference of normal `Parse.Cloud` syntax `Parse.Cloud.beforeDelete(Parse.File` | [#7927](https://github.com/parse-community/parse-server/pull/7927)   |6.0.0 (2022)                    | 6.0.0 (2023)                    | deprecated            | -     |
```
Comment (mtrezza) [inline DEPRECATIONS.md:11]: The deprecation would happen in 6.0 (2023), removal in 7.0 (2024).

Typo, missing closing bracket.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
> This can only be used on Parse.File for the meantime, but in the future could be added to detect class creation. 

I'd say the equivalent of Parse.File creation would be a Parse.Object creation, not a class creation. I can already imagine some use cases for that.

**Comment by dblythy** [comment]
> I'd say the equivalent of Parse.File creation would be a Parse.Object creation

Well I think a Parse.Object creation can be covered by wild class beforeSave triggers. In any case, it's out of scope of this PR. I personally think how the other `beforeCreate` triggers are implemented should be up for its own discussion.

File: DEPRECATIONS.md (hunk 1 above)
Comment (dblythy) [reply DEPRECATIONS.md:11]: Yes, as to my other PRs i'm not sure how to fill this table. How can an item's status be "Deprecated" if its planned Deprecation is Version 6?

**Comment by mtrezza** [comment]
The naming discussion is part of this PR because you are introducing a new trigger type. We want to find a name that is universal enough to be applicable to classes in the future. So let's think a few steps ahead...

In my understanding, `beforeCreate` is different from a `beforeSave`, just as creating/instantiating an object is different from saving it.

We may also soon have an internal `File` class as was discussed related to a few features. The `beforeCreate` trigger won't be triggered when the `File` class is created, nor when a File object is saved. A developer would expect the same behavior when the trigger is applied to other classes.

Maybe we should remove / rephrase the assumption that the will be related to class creation from the PR description to not confuse future readers about this?

**Comment by dblythy** [comment]
Maybe the discussion for a beforeCreate trigger should happen elsewhere. I think it’s relevant to our discussions to have a trigger to mutate the file url. The name of that trigger might be better off with something else if you feel `beforeCreate` doesn’t represent its functions.

**Comment by mtrezza** [comment]
> if you feel beforeCreate doesn’t represent its functions

The question is what would be "its function" when used with classes? I think `beforeCreate` is fine assuming that it can be used with classes for when an object is created, not for when a class is created. To know whether the new trigger name is well chosen in the sense that it can be repurposed for classes, we need to think ahead about what it could do when used with classes.

**Comment by dblythy** [comment]
Closing - will reopen later

**Comment by dblythy** [comment]
So you think `beforeCreate` should effectively be a beforeSave trigger but for !object.existed()? How would that affect the existing beforeSave trigger?

I also closed this and created #7966 to split it into smaller PRs. I will create a PR relating to this feature.

**Comment by mtrezza** [comment]
I would use `beforeCreate` to trigger when a `Parse.Object` instantiates.

1. `new Parse.Object()` -> `beforeCreate`, for example to set default properties of an object
2. `Parse.Object.save()` -> `beforeSave`, `afterSave`

You wrote:

> This triggers when a file is expanded, allowing mutation of the file url.

What did you mean by "expanded"? What issue were you intending to solve with the `beforeCreate(Parse.File)` trigger?

**Pull request closed without merging**